modules/mcc/src/checker_detector.cpp

```diff
@@ -0,0 +1,1208 @@
+		// Przegl?d Elektrotechniczny, 90(9), 197-202.
+		// 0.65 < \frac{4*pi*A}{P^2} < 0.97
+		// double Cf = 4 * CV_PI * area / (perm * perm);
+		// if (Cf < 0.5 || Cf > 0.97) continue;
+
+		// Soliditys
+		// This measure is proposed in this work.
+		PointsVector hull;
+		cv::convexHull(contour, hull);
+		double area_hull = cv::contourArea(hull);
+		double S = area / area_hull;
+		if (S < params->minContourSolidity)
+			continue;
+
+		// Texture analysis
+		// ...
+
+		contours.push_back(allContours[i]);
+	}
+}
+
+void CCheckerDetectorImpl::
+	findCandidates(
+		const ContoursVector &contours,
+		std::vector<CChart> &detectedCharts,
+		const Ptr<DetectorParameters> &params)
+{
+	std::vector<cv::Point> approxCurve;
+	std::vector<CChart> possibleCharts;
+
+	// For each contour, analyze if it is a parallelepiped likely to be the chart
+	for (size_t i = 0; i < contours.size(); i++)
+	{
+		// Approximate to a polygon
+		//  It uses the Douglas-Peucker algorithm
+		// http://en.wikipedia.org/wiki/Ramer-Douglas-Peucker_algorithm
+		double eps = contours[i].size() * params->findCandidatesApproxPolyDPEpsMultiplier;
+		cv::approxPolyDP(contours[i], approxCurve, eps, true);
+
+		// We interested only in polygons that contains only four points
+		if (approxCurve.size() != 4)
+			continue;
+
+		// And they have to be convex
+		if (!cv::isContourConvex(approxCurve))
+			continue;
+
+		// Ensure that the distance between consecutive points is large enough
+		float minDist = INFINITY;
+
+		for (size_t j = 0; j < 4; j++)
+		{
+			cv::Point side = approxCurve[j] - approxCurve[(j + 1) % 4];
+			float squaredSideLength = (float)side.dot(side);
+			minDist = std::min(minDist, squaredSideLength);
+		}
+
+		// Check that distance is not very small
+		if (minDist < params->minContourLengthAllowed)
+			continue;
+
+		// All tests are passed. Save chart candidate:
+		CChart chart;
+
+		std::vector<cv::Point2f> corners(4);
+		for (int j = 0; j < 4; j++)
+			corners[j] = cv::Point2f((float)approxCurve[j].x, (float)approxCurve[j].y);
+		chart.setCorners(corners);
+
+		possibleCharts.push_back(chart);
+	}
+
+	// Remove these elements which corners are too close to each other.
+	// Eliminate overlaps!!!
+	// First detect candidates for removal:
+	std::vector<std::pair<int, int>> tooNearCandidates;
+	for (size_t i = 0; i < possibleCharts.size(); i++)
+	{
+		const CChart &m1 = possibleCharts[i];
+
+		//calculate the average distance of each corner to the nearest corner of the other chart candidate
+		for (size_t j = i + 1; j < possibleCharts.size(); j++)
+		{
+			const CChart &m2 = possibleCharts[j];
+
+			float distSquared = 0;
+
+			for (int c = 0; c < 4; c++)
+			{
+				cv::Point v = m1.corners[c] - m2.corners[c];
+				distSquared += v.dot(v);
+			}
+
+			distSquared /= 4;
+
+			if (distSquared < params->minInterContourDistance)
+			{
+				tooNearCandidates.push_back(std::pair<int, int>(i, j));
```

Comment:
`i` and `j` are `size_t` and you are automatically casting to int. This will generate failing warning in the CI

Reply:
Thanks a lot! I was trying to fix it but couldn't pinpoint what was causing this warning.